### ISSUE TITLE
Allow parameter to specify the filename for "the main template"

### ISSUE THREAD

**Issue by JFolberth** [Azure/arm-ttk#96]
In the location test for the TTK there is a flag to check if the file is the $IsMainTemplate. This boolean is if the file name is one of 'mainTemplate.json', 'azureDeploy.json', 'prereq.azuredeploy.json'. If so then check the parameter location for the correct definition of resourceGroup.location() being the default.

This prohibits any deployment files from being named differently. 

Here is the code for the test:
````
if ($IsMainTemplate){ 
    if($locationParameter.defaultValue -and "$($locationParameter.defaultvalue)".Trim() -ne '[resourceGroup().location]') {
    Write-Error "The defaultValue of the location parameter in the main template must not be a specific location. The default value must be [resourceGroup().location]. It is 
`"$($locationParameter.defaultValue)`"" -ErrorId Location.Parameter.Hardcoded -TargetObject $parameter
}
# In all other templates:
# if the parameter named "location" exists, it must not have a defaultValue property
# Note that Powershell will count an empty string (which should fail the test) as null if not explictly tested, so we check for it
}else {
    if($locationParameter.defaultValue -ne $null){ 
        Write-Error "The location parameter of nested templates must not have a defaultValue property. It is `"$($locationParameter.defaultValue)`"" -ErrorId Location.Parameter.DefaultValuePresent -TargetObject $parameter
    }   
}

This is problematic as if deploying multiple arm templates it isn't always ideal to be naming the same.

**Comment by bmoore-msft** [comment]
This is intentional...  the scenario is this:

If the file is the top level deployment template, then it's permissible to provide a defaultValue on the location parameter.  If the file is a nested template, then the value of location must be passed through.  If there is a defaultValue, then it's easy to forget to pass the value through.  We check on the nested template (instead of the invocation of the template in the deployments resource) because it's much easier.

Now re: the naming... it's conceivable to have a param that indicates the name of mainTemplate - so if your "top level template" is called foo.json, you could pass something to the tests to indicate that you don't have a mainTemplate.json (for example) but it's called "foo.json".

If that sounds right to you, we can change the title of this issue to reflect that...

**Comment by JFolberth** [comment]
That sounds potentially good to me.

**Comment by bmoore-msft** [comment]
Fixed via #261
